api/controllers/ResumeController.ts

```diff
@@ -0,0 +1,25 @@
+import { JsonController, Params, Get, UseBefore } from 'routing-controllers';
+import PermissionsService from 'services/PermissionsService';
+import { UserAuthentication } from '../middleware/UserAuthentication';
+import ResumeService from '../../services/ResumeService';
+import { UserModel } from '../../models/UserModel';
+import { UuidParam } from '../validators/GenericRequests';
+import { AuthenticatedUser } from '../decorators/AuthenticatedUser';
+import { GetResumesListResponse } from '../../types';
+
+@UseBefore(UserAuthentication)
+@JsonController('/resume')
+export class ResumeController {
+  private resumeService: ResumeService;
+
+  constructor(resumeService: ResumeService) {
+    this.resumeService = resumeService;
+  }
+
+  @Get()
+  async getAllVisibleResumes(@Params() params: UuidParam,
+    @AuthenticatedUser() currentUser: UserModel): Promise<GetResumesListResponse> {
+    const resumes = await this.resumeService.getVisibleResumes(PermissionsService.canSeeAllVisibleResumes(currentUser));
+    return { error: null, resumes };
+  }
+}
```

Comment:
We should throw if someone does not have permission to see all resumes (and if someone doesn't have permission to view stuff in general). If someone who doesn't have permission is calling this API route, this is fundamentally an error by the caller, and is something that we should throw a 401 Forbidden Error for. Rerturning an empty response makes it seem like they _do_ have permission, but that there just aren't any resumes, which is incorrect from a data point of view.

Reply:
Would you recommend handling the permission error in the controller in this case?

Reply:
Yes, since permissions are related to whether or not we should process a request, permission logic should happen in the controller